pandas/core/frame.py

```diff
@@ -369,6 +369,10 @@ class DataFrame(NDFrame):
     2  7  8  9
     """
 
+    if TYPE_CHECKING:  # attributes index and columns are created dynamically
+        index = None  # type: Index
+        columns = None  # type: Index
```

Comment:
i think the dynamic creation is an artifact of PanelND support, could probably be hard-coded now.

Index is definitely them user-correct type, but I think it might be AxisProperty internally

Reply:
> could probably be hard-coded now.

I agree. (but not in this PR).

> but I think it might be AxisProperty internally

not sure it this will make a difference. clean mypy build for now though. Thanks for highlighting this potential issue. May surface as add more type hints to core.generic

Reply:
How complex is that refactor? Seems like this would be a logical time to do that so types *could* resolve implicitly rather than having to add more type hinting logic here